ENIGMAsystem/SHELL/Platforms/Win32/WINDOWSexternals.cpp

```diff
@@ -154,24 +154,26 @@ variant external_call(int id,variant a1,variant a2, variant a3, variant a4, vari
   map<int,external*>::iterator it;
   if ((it=externals.find(id)) == externals.end())
   {
-  	show_error("Unknown external function called", 0);
-  	return 0;
+    show_error("Unknown external function called", 0);
+    return 0;
   }
   external* a=it->second;
-  
+
   ambiguous array[a->argc];
   void *arg_values[a->argc];
-  
+
   variant args[] = { a1, a2, a3, a4, a5, a6, a7, a8, a9, a10, a11, a12, a13, a14, a15, a16 };
-  for (int i=0;i<a->argc;i++)
+  for (int i = 0; i < a->argc; ++i)
   {
-    if (a->arg_type[i]==&ffi_type_double)
-      array[i].d=(double)args[i];
+    if (a->arg_type[i] == &ffi_type_double)
+      array[i].d = (double)args[i];
```

Comment:
This is messy, because we might send the double-encoded version of a pointer, here. It's probably the best strategy, though, since doing a "proper" cast could cause the double to lose precision, anyway.

Reply:
Fixed in commit 0c602f67dff817b7e596fa318e09048ce2696bec

Reply:
Really, shouldn't this function have been expanded to accept ty_int32, ty_int64, etc? That would have avoided all of this implicit conversion, I don't know why they did it this way instead of adding all of the type constants. They probably felt that more than 2 types was too advanced for GM users, even though they think all this casting madness is trivial.